### DISCUSSION TITLE
Suggestion on howto handle Jobs?

### DISCUSSION THREAD

**Discussion by strowi** [grafana/tanka#727] [Q&A] [answered]
Hi,

i am currently testing tanka on a project involving the [bitnami keycloak helm-chart](https://bitnami.com/stack/keycloak/helm), but this applies to all jobs i guess.
When deploying keycloak it deploys also a kubernetes-job to initalize the realms etc.
Now when this has run (successful or not), i have to delete the job manually to be able to re-deploy and run that job.

Anyone have a suggestion on how to better handle this?

regards,
strowi

**Comment by BeyondEvil** [comment]
How is the job deployed? Manually? Using something like Github Actions? ArgoCD?

**Comment by julienduchesne** [comment, marked as answer]
You can add a random string to the job name depending on parameters (keycloak version?). We have this for Flux (deploying Flux with Flux):
```
  } + {
      // Hash the whole job object and add the hash to its name
      // The job's template is immutable, so this ensures that any modification recreates it completely.
      local hash = std.md5(std.toString(super.job_flux2_flux_check)),
      job_flux2_flux_check+: k.batch.v1.job.metadata.withName('flux2-check-' + hash),
    },
```

**Reply by Duologic** [reply]
I did something similar, the job was configured through a configMap, the Job name was suffixed with an `std.md5` of that.
So after the name change, Tanka can prune the previous version, or if you're using a CD tool like Flux then previous versions automatically gets cleaned up.

**Comment by strowi** [comment]
Thanks for all the answers. This is deployed manally for testing and via gitlab in prod. I went with adding an md5 hash of the config as suggested by @julienduchesne . That doesn't factor in changes to the job itself (like volumeMounts etc) but it is good enough for now.;)

PS: In prod via ci-job it is definitely possible to prepend a `kubectl delete jobs --all` or similar.
